### PR TITLE
Docs: Update semantic versioning policy

### PR DESCRIPTION
**Pull request by nzakas**
Explicitly mention how we handle new language features for rules.

#### Prerequisites checklist

- [ ] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Updated the semantic versioning policy per our discussion on 2020-12-31 to explicitly state that rules can add new language features that result in more linting errors by default in a semver-minor release.

#### Is there anything you'd like reviewers to focus on?

Did I word this addition correctly?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -158,6 +158,7 @@ ESLint follows [semantic versioning](https://semver.org). However, due to the na
     * A bug fix in a rule that results in ESLint reporting more linting errors.
     * A new rule is created.
     * A new option to an existing rule that does not result in ESLint reporting more linting errors by default.
+    * A new option to an existing rule to support a newly-added language feature (within the last 12 months) that will result in ESLint reporting more linting errors by default.
```
Comment (mdjermanovic) [inline README.md:161]: It doesn't have to be an option?

Comment (nzakas) [reply README.md:161]: Maybe “addition” is a better word?

Comment (mdjermanovic) [reply README.md:161]: Sounds good to me!

**Review by btmills — COMMENTED** [review]
LGTM once clarified as "addition".

File: README.md (hunk 1 above)
Comment (nzakas) [reply README.md:161]:
```suggestion
    * A new addition to an existing rule to support a newly-added language feature (within the last 12 months) that will result in ESLint reporting more linting errors by default.
```

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**